ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/cache/LRUCache.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.datastructures.util.cache;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Cache made around LRU-map with fixed size, removing eldest entries (by added) when the space is
+ * over
+ *
+ * @param <K> Keys type
+ * @param <V> Values type
+ */
+public class LRUCache<K, V> implements Cache<K, V> {
+
+  private final Map<K, V> cacheData;
+  private final int maxCapacity;
+
+  /**
+   * Creates cache
+   *
+   * @param capacity Size of the cache
+   */
+  public LRUCache(int capacity) {
+    this(capacity, Collections.emptyMap());
+  }
+
+  private LRUCache(int capacity, Map<K, V> initialCachedContent) {
+    this.maxCapacity = capacity;
+    this.cacheData =
+        Collections.synchronizedMap(
+            new LinkedHashMap<>(maxCapacity + 1) {
```

Comment:
Could this just use `LimitedHashMap`?  The static block to add initialCachedContent would just have to be done after construction but that should be safe since we're still in the constructor so no other operations can be happening.

Reply:
Yep 👍 
Done: https://github.com/PegaSysEng/artemis/pull/1111/commits/19d72bc794a3046bed74b5f9380a3bff80835604